gef.py

```diff
@@ -10518,13 +10522,12 @@ def read_ascii_string(self, address: int) -> Optional[str]:
     @property
     def maps(self) -> List[Section]:
         if not self.__maps:
-            self.__maps = self._parse_maps()
+            self.__maps = self.__parse_maps()
             if not self.__maps:
                 raise RuntimeError("Failed to get memory layout")
         return self.__maps
 
-    @classmethod
-    def _parse_maps(cls) -> Optional[List[Section]]:
+    def __parse_maps(self) -> Optional[List[Section]]:
```

Comment:
Why change this? It breaks the ability for a subclass to class it.

Reply:
I usually avoid exposing to subclass unless purposely designed and here I'm not sure I see a situation where we'd make a subclass but *still* use the same memory parsing functions. My logic is that, if you have the need to make a subclass you already need another way to parse memory (from a different source).
I don't feel strongly about this so if you can think of such case, lmk and I'll revert

Reply:
Well this function makes sense for subclasses to use, it's pretty standardized. And even if this is `_parse_maps` we still have the _option_ to override it. All you're doing is taking away the ability of a subclass to use it. I actually removed the double underscore in a previous PR because it was hurting me. `__` methods are actually meant to prevent accidentally overridding.

This `__parse_maps` is not as bad. But say that I want to write my own `__parse_maps`, and I want to change the ORDER of the different attempts. I can't (easily) do this, since you're hiding all the methods! This breaks the blackmagicprobe support, for example: https://github.com/hugsy/gef-extras/blob/main/archs/arm-blackmagicprobe.py#L29

Reply:
Got it, makes sense. I'll revert to protected 👍